### DISCUSSION TITLE
Update HSL color variables to modern syntax?

### DISCUSSION THREAD

**Discussion by thdoan** [saadeghi/daisyui#1771] [Ideas / feature requests]
[MDN](https://developer.mozilla.org/docs/Web/CSS/color_value/hsl) states that the `hsla()` syntax daisyUI is using is "legacy":

```
/* Legacy hsla() syntax */
hsla(hue saturation lightness / alpha)
```

Not sure what this means down the road, but maybe start to think about updating HSL variables to the modern syntax? If anything this would make it consistent with `rgba()` syntax.

Example:
```
:root {
  --pc: 210, 100%, 86.118%;
}

.card {
  background-color: hsla(var(--pc), 0.8);
}
```

**Comment by saadeghi** [comment]
This wouldn't cause any issues. It's valid style and it works on all browsers.  
>The legacy `hsla()` syntax is an alias for `hsl()`

But anyway it will be fixed in daisyUI 3.0 because I'm updating the color functions.  
In daisyUI 3.0 color values will be LCH by default with a HSL fallback. this allows us to use P3 colors while providing a fall back for older browsers.
